framework/render/tasks.py

```diff
@@ -46,6 +47,36 @@ def render_mfr_error(err):
         """.format(**locals())
 
 
+@app.task(ignore_result=True, timeout=settings.MFR_TIMEOUT)
+def get_file_contents(download_url, cache_path, temp_path, public_download_url):
+
+    ensure_path(os.path.split(temp_path)[0])
+    ensure_path(os.path.split(cache_path)[0])
+
+    try:
+        save_to_file_or_error(download_url, temp_path)
+    except exceptions.RenderNotPossibleException as e:
+        # Write out unavoidable errors
+        content = e.renderable_error
+    else:
+        encoding = None
+        if get_file_extension(temp_path) in CODE_EXTENSIONS:
+            encoding = 'utf-8'
+        with codecs.open(temp_path, encoding=encoding) as temp_file:
+            try:
+                rendered_html = mfr.render(fp=temp_file, src=public_download_url).content
+                soup = BeautifulSoup(rendered_html)
```

Comment:
Please move these two lines not caught by the except block out of this try block. I.e.:
```
    # Rendered MFR error
    content = render_mfr_error(err)
soup = BeautifulSoup(rendered_html)
content = str(soup.get_text())
```